students/mielczarek_szymon/lesson_04_unit_testing/test_distance.py

```diff
@@ -0,0 +1,36 @@
+import sys
+import unittest
+from lesson_03_functions.the_length_of_the_segment import distance
+sys.path.append('..')
+
+
+class TestDistance(unittest.TestCase):
+    def test_on_none(self):
+        self.assertRaises(TypeError, distance, None, 1, 2, 3)
+
+    def test_non_integer(self):
+        self.assertRaises(TypeError, distance, 'aoeu', 1, 2, 3)
+
+    def test_zero_length(self):
+        self.assertEqual(distance(2, 2, 2, 2), 0)
+
+    def test_negative_coordinates(self):
+        self.assertEqual(distance(-2, -7, -13, -1), 12.529964086141668)
+
+    def test_vertical_distance(self):
+        self.assertEqual(distance(2, 2, 2, 12), 10)
+
+    def test_horizontal_distance(self):
+        self.assertEqual(distance(2, 2, 12, 2), 10)
+
+    def test_typical_coordinates(self):
+        self.assertEqual(distance(2, 3, 12, 4), 10.04987562112089)
```

Comment:
I know it wasn't specified, but there is no test scenario with float numbers as input

Reply:
But there is no float inputs distance(2, 3, 12, 4), only outputs :)